repository/cat/cat_repository_impl.go

```diff
@@ -116,3 +116,33 @@ func (repository *CatRepositoryImpl) Search(ctx context.Context, tx pgx.Tx, sear
 
 	return cats, nil
 }
+func (repository *CatRepositoryImpl) Edit(ctx context.Context, tx pgx.Tx, cat cat_entity.Cat, catId string, ownerId string) (cat_entity.Cat, error) {
+	//  first we check whether the cat is already match(someone request to match) or not(no catfishing, pun intended)
+	var matchCatSex string
+	match_query := `Select cats.sex from matches join cats on cats.id=$1 where cat_issuer_id=$1 or cat_receiver_id=$1 and status!='rejected';`
+	tx.QueryRow(ctx, match_query, catId).Scan(&matchCatSex)
```

Comment:
jangan lupa handle error ya bangg

Reply:
ini gw masih bingung dah bang, soalnya kalau errornya not found gitu emang mau gw skip, hmm atau gini aja  ya


```go
err = tx.QueryRow(ctx, match_query, catId).Scan(&matchCatSex)
if !strings.Contains(err.Error(), "no rows in result set") {
  		return cat_entity.Cat{}, err
}


```

Reply:
ini gw masih bingung dah bang, soalnya kalau errornya not found gitu emang mau gw skip, hmm atau gini aja  ya


```go
err = tx.QueryRow(ctx, match_query, catId).Scan(&matchCatSex)
if !strings.Contains(err.Error(), "no rows in result set") {
  		return cat_entity.Cat{}, err
}


```